stl/inc/ranges

```diff
@@ -728,6 +735,441 @@ namespace ranges {
+                const _Iterator& _Left, const _Iterator& _Right) requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Left._Compat(_Right);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _Left._Current < _Right._Current;
+            }
+            _NODISCARD friend constexpr bool operator>(
+                const _Iterator& _Left, const _Iterator& _Right) requires random_access_range<_Base> {
+                return _Right < _Left;
+            }
+            _NODISCARD friend constexpr bool operator<=(
+                const _Iterator& _Left, const _Iterator& _Right) requires random_access_range<_Base> {
+                return !(_Right < _Left);
+            }
+            _NODISCARD friend constexpr bool operator>=(
+                const _Iterator& _Left, const _Iterator& _Right) requires random_access_range<_Base> {
+                return !(_Left < _Right);
+            }
+            // clang-format off
+            _NODISCARD friend constexpr auto operator<=>(const _Iterator& _Left, const _Iterator& _Right)
+                requires random_access_range<_Base> && three_way_comparable<iterator_t<_Base>> {
+                // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Left._Compat(_Right);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _Left._Current <=> _Right._Current;
+            }
+
+            _NODISCARD friend constexpr _Iterator operator+(
+                _Iterator _It, difference_type _Off) requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _It._Verify_offset(_Off);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                _It._Current += _Off;
+                return _It;
+            }
+            _NODISCARD friend constexpr _Iterator operator+(
+                difference_type _Off, _Iterator _It) requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _It._Verify_offset(_Off);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                _It._Current += _Off;
+                return _It;
+            }
+
+            _NODISCARD friend constexpr _Iterator operator-(
+                _Iterator _It, difference_type _Off) requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _It._Verify_offset(-_Off);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                _It._Current -= _Off;
+                return _It;
+            }
+
+            _NODISCARD friend constexpr difference_type operator-(
+                const _Iterator& _Left, const _Iterator& _Right) requires random_access_range<_Base> {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Left._Compat(_Right);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _Left._Current - _Right._Current;
+            }
+
+            _NODISCARD friend constexpr decltype(auto) iter_move(const _Iterator& _It) noexcept(
+                noexcept(_STD invoke(*_It._Parent->_Fun, *_It._Current))) {

```

Comment:
That noexcept clause here looks strange

Reply:
Ha! I tried to convince LWG that we should make `operator*` be `noexcept(noexcept(invoke(*_Parent->_Fun, *_Current)))` so this could simply say `noexcept(*_It)` because everyone who read this wanted an explanation. I couldn't do so in the Standard, but I can here.